### ISSUE TITLE
warnings when opening a terminal 

### ISSUE THREAD

**Issue by gozes** [fisherman/fzf#7]
I get the following warnings whenever I open a shell. I looked inside the file and it belongs to fzf so that is way I am opening the issue here 

```shell 
fish: Unknown command '__fisher_key_bindings'
~/.config/fish/functions/fish_user_key_bindings.fish (line 1): __fisher_key_bindings
                                                               ^
in function “fish_user_key_bindings”
	called on standard input

in function “__fish_reload_key_bindings”
	called on line 183 of file /usr/share/fish/functions/__fish_config_interactive.fish

in function “__fish_config_interactive”
	called on line 137 of file /usr/share/fish/config.fish

in function “__fish_on_interactive”
	called on standard input

in event handler: handler for generic event “fish_prompt”
```

**Comment by jethrokuan** [comment]
This is likely because of the incompatible upgrade of fisherman from 1.5 to
2.x.

Unfortunately there is no clean way to resolve this. The short answer is
the keybindings solution in 2.x is different from that in 1.x, and does not
require the intermediary __fisher_key_bindings variable.

You'd need to uninstall fzf, then open user_key_bindings.fish and remove
the reference to__fisher_key_bindings (or delete the file altogether if
nothing else is in there), then reinstall fzf.

**Comment by jethrokuan** [comment]
@gozes I just got back from a holiday. Are you still facing any issues?

**Comment by gozes** [comment]
yes I am. I didn't reinstall or anything because I think that that this is something that should be handle upstream(here or in fisher not sure where) because if  I do so that would mean that I have a "fork" that is unique  in some way to upstream that would mean that fisher update could break or just not work because of changes. sorry if it sounds to dramatic but there should  be clear divide between what is user configurable stuff that is meant to modified by the user and what is the system that  user dose not have to touch unless you are hunting a bug down or are planing to contribute to the project. 

I hope you enjoy your holiday :smile: :+1: :palm_tree:

**Comment by jethrokuan** [comment]
@gozes I think I didn't phrase myself correctly.

`__fisher_key_bindings` was how **fisherman** - not this fzf plugin - handled key bindings.

There was a breaking change in fisherman 2.0 that removed `__fisher_key_bindings` entirely, in place of a different key-binding solution. Therefore, if you have downloaded fisherman 1.x and upgraded it to 2.x **without** uninstalling and reinstalling fzf, it **will** break.

The reason you're getting this error is because the migration path from 1.x to 2.x introduced breaking changes, and there is nothing I can do to help make that easier.

TL;DR this is a fisherman migration issue, not a fzf issue. The above fix I mentioned should work.

**Comment by jethrokuan** [comment]
Perhaps @brj can provide a better explanation

**Comment by brj** [comment]
@gozes Run fisher self-uninstall and then install fisherman again. Then, the problem will likely go away.

**Comment by gozes** [comment]
Will all the plugins I have install will be uninstall as well or will i
keep my fisherfile?

**Comment by brj** [comment]
@gozes This is a special case, because of the problematic 1.5 to 2.0 upgrade (that I believe is the case here), so what I recommend this time is you make a copy of your fishfile, then proceed to completely remove fisherman using the method described above and then just install again using the simpler:

```
curl -Lo ~/.config/fish/functions/fisher.fish --create-dirs git.io/fisher
```

**Comment by jethrokuan** [comment]
@gozes is the fix alright for you? If so, feel free to close the issue.

**Comment by gozes** [comment]
I will test it when i get back. I don't have my system where i am at the
moment.

**Comment by jethrokuan** [comment]
Closing as answered/resolved.
